src/RTT_Format_Reader/RTT_Format_Reader.cc

```diff
@@ -17,12 +17,12 @@ namespace rtt_RTT_Format_Reader {
  * \brief Constructs an RTT_Format_Reader object and parses the mesh data.
  * \param RTT_File Mesh file name.
  */
-RTT_Format_Reader::RTT_Format_Reader(string const &RTT_File)
+RTT_Format_Reader::RTT_Format_Reader(string RTT_File)
     : header(), dims(), spNodeFlags(), spSideFlags(), spCellFlags(),
       spNodeDataIds(), spSideDataIds(), spCellDataIds(), spCellDefs(),
       spNodes(), spSides(), spCells(), spNodeData(), spSideData(),
       spCellData() {
-  readMesh(RTT_File);
+  readMesh(std::move(RTT_File));
```

Comment:
I'm not up on the intricacies of moves and strings. This is a copy followed by a move?

Reply:
That is correct. In this scenario, a copy must be made and the recommendation is to force the copy in the constructor function call reducing the number of steps required.  Gritty details are [here](https://releases.llvm.org/8.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/modernize-pass-by-value.html)